### PR TITLE
arch-riscv: Fix c.jalr and c.jr instruction

### PR DESCRIPTION
**Pull request by rogerchang23424**
The bit 0 of register should be 0 for jump address. Wrong handling the jump address may cause infinite run or segment fault.

gem5 issue: https://github.com/gem5/gem5/issues/981
Change-Id: I605de64bd3cece70288ffcfaf1f4263c17ab2971

### CODE REVIEW

File: src/arch/riscv/isa/formats/compressed.isa [hunk 1]
```diff
@@ -111,6 +111,8 @@ def format CompressedStore(sdisp_code, memacc_code,
         inst_flags, 'Store', exec_template_base='Store')
 }};
 
+
+
```
Comment (hnpl) [inline src/arch/riscv/isa/formats/compressed.isa]: Nit: these extra empty lines should be removed.

**Review by hnpl — APPROVED** [review]
The code looks good to me.

I'm not sure if the riscv-related tests we have contain those instructions. Can you let me know if the code has been tested?

**Comment by rogerchang23424** [comment]
I have tested them with riscof. However, the case doesn't included in the riscof, so I add them to test case https://github.com/riscv-non-isa/riscv-arch-test/pull/466.

Since we can't dump the memory signature directly with gem5. Instead, I compared the signature with customized plugin (https://github.com/gem5/gem5-resources/pull/7) with other simulators.

**Comment by hnpl** [comment]
That's great!!

**Review by BobbyRBruce — APPROVED** [review]

**Pull request merged**